libpod/healthcheck.go

```diff
@@ -236,6 +236,11 @@ func (c *Container) HealthCheckStatus() (string, error) {
 	if !c.HasHealthCheck() {
 		return "", errors.Errorf("container %s has no defined healthcheck", c.ID())
 	}
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if err := c.syncContainer(); err != nil {
+		return "", err
+	}
 	results, err := c.GetHealthCheckLog()
```

Comment:
Should we just change `GetHealthCheckLog()` to do the lock/sync bits instead, to make it safe for all callers?

Reply:
I believe the container is already locked in the other code paths which use GetHealthCheckLog().

Reply:
We should make it a private function then, so nobody calls it and gets nulls

Reply:
done